### PR TITLE
Fixes #2260 fixedProfileCrash added permission 

### LINKED ISSUE #2260: app crash when pickingUp image from gallery
**Actual Behaviour**

App crashes when image is picked for profile

**Expected Behaviour**

App should not crash

**Steps to reproduce it**
go to login
click profile logo
pick a image 
tada app crashed .....

**LogCat for the issue**

`java.lang.RuntimeException: An error occurred while executing doInBackground()
 at android.os.AsyncTask$3.done(AsyncTask.java:325)
 at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
 at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
 at java.util.concurrent.FutureTask.run(FutureTask.java:242)
 at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
 at java.lang.Thread.run(Thread.java:760)
Caused by: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/images/media from pid=22827, uid=10533 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
 at android.os.Parcel.readException(Parcel.java:1684)
 at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
 at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
 at android.content.ContentProviderProxy.query(ContentProviderNative.java:421)
 at android.content.ContentResolver.query(ContentResolver.java:534)
 at android.content.ContentResolver.query(ContentResolver.java:475)
 at com.yalantis.ucrop.util.FileUtils.getDataColumn(FileUtils.java:109)
 at com.yalantis.ucrop.util.FileUtils.getPath(FileUtils.java:183)
 at com.yalantis.ucrop.task.BitmapLoadTask.processInputUri(BitmapLoadTask.java:171)
 at com.yalantis.ucrop.task.BitmapLoadTask.doInBackground(BitmapLoadTask.java:90)
 at com.yalantis.ucrop.task.BitmapLoadTask.doInBackground(BitmapLoadTask.java:41)
 at android.os.AsyncTask$2.call(AsyncTask.java:305)
 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
 ... 4 more`

**Screenshots of the issue**

ezgif com-video-to-gif 3

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by jindalAnuj**
Fixes #2260 . app crash when pickingUp image from gallery e.g. #2260

Changes: [Permission added for picking image
added

and some code for checking android version in editprofileActivity

### CODE REVIEW

**Comment by jindalAnuj** [comment]
@iamareebjamal please review this PR

File: android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java [hunk 1]
```diff
@@ -82,14 +87,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    }
+    private boolean checkIfAlreadyhavePermission() {
+        int result = ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE);
+        return result == PackageManager.PERMISSION_GRANTED;
+
+    }
+    private void requestForSpecificPermission() {
+        ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 101);
+    }
+
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java:117]: Several syntax and style issues. Please resolve

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java [hunk 2]
```diff
@@ -82,14 +87,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
     }
 
+    private void androidVersionCheck()
+    {
+        int MyVersion = Build.VERSION.SDK_INT;
+        if (MyVersion > Build.VERSION_CODES.LOLLIPOP_MR1) {
+            if (!checkIfAlreadyhavePermission()) {
+                requestForSpecificPermission();
+            }
+            else showFileChooser();
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java:105]: Don't we want to show the File chooser after requesting for permissions ?

File: android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java [hunk 3]
```diff
@@ -82,14 +87,34 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        avatar.setOnClickListener(v -> showFileChooser());
+        avatar.setOnClickListener(v ->  androidVersionCheck());
 
         save.setOnClickListener(v -> {
             Views.hideKeyboard(this, this.getCurrentFocus());
             saveChanges();
         });
     }
 
+    private void androidVersionCheck()
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java:98]: Rename it to ```checkAndroidVersion```

**Review by dr0pdb — CHANGES_REQUESTED** [review]
Fix the formatting errors also.

**Comment by iamareebjamal** [comment]
@jindalAnuj Status?

**Pull request closed without merging**